server/src/test/java/org/elasticsearch/cluster/metadata/MetadataTests.java

```diff
@@ -1078,6 +1080,148 @@ public void testSerialization() throws IOException {
         assertTrue(Metadata.isGlobalStateEquals(orig, fromStreamMeta));
     }
 
+    public void testValidateDataStreamsNoConflicts() {
+        Metadata metadata = createIndices(5, 10, "foo-datastream").metadata;
+        try {
+            validateDataStreams(metadata.getIndicesLookup(), (DataStreamMetadata) metadata.customs().get(DataStreamMetadata.TYPE));
```

Comment:
Perhaps just add a comment on why no failure is expected and let the exception bubble up?

Reply:
Is the aim of this change a better runtime experience in case the test fails? Or test readability?

I think the `fail` message is providing the runtime context, and it will print it in case of failure (which should make debugging this straightforward).

Reply:
Currently when the test fails with any error then there is no stacktrace to the original failure, which I think makes pinpointing the culprit easier?

That being said, this is just preference, so feel free to ignore :)

Reply:
TIL didn't realise it hides the cause of the failure. Made the change, thanks!